server-timing/load.php

```diff
@@ -113,3 +116,66 @@ function perflab_wrap_server_timing( $callback, $metric_slug, $access_cap ) {
 		return $result;
 	};
 }
+
+/**
+ * Registers the Server-Timing setting.
+ *
+ * @since n.e.x.t
+ */
+function perflab_register_server_timing_setting() {
+	register_setting(
+		PERFLAB_SERVER_TIMING_SCREEN,
+		PERFLAB_SERVER_TIMING_SETTING,
+		array(
+			'type'              => 'object',
+			'sanitize_callback' => 'perflab_sanitize_server_timing_setting',
+			'default'           => array(),
+		)
+	);
+}
+add_action( 'init', 'perflab_register_server_timing_setting' );
+
+/**
+ * Sanitizes the Server-Timing setting.
+ *
+ * @since n.e.x.t
+ *
+ * @param mixed $value Server-Timing setting value.
+ * @return array Sanitized Server-Timing setting value.
+ */
+function perflab_sanitize_server_timing_setting( $value ) {
+	if ( ! is_array( $value ) ) {
+		return array();
+	}
+
+	/*
+	 * Ensure that every element is an indexed array of hook names.
+	 * Any duplicates across a group of hooks are removed.
+	 */
+	return array_filter(
+		array_map(
+			static function( $hooks ) {
+				if ( ! is_array( $hooks ) ) {
+					$hooks = explode( "\n", $hooks );
+				}
+				return array_values(
+					array_unique(
+						array_filter(
+							array_map(
+								static function( $hookname ) {
+									return preg_replace(
+										'/[^A-Za-z0-9_\-\/\.]/',
+										'',
+										strtolower( $hookname )
```

Comment:
Hooks can involve upper-case letters: https://wpdirectory.net/search/01H78FM13RS16AKX0G7K06N2JE

```suggestion
										$hookname
```

Reply:
Also, probably good to add `trim()`:

```suggestion
										trim( $hookname )
```

Reply:
See my reply in https://github.com/WordPress/performance/pull/784#discussion_r1286215594.